changefeedccl: pause schedules when restored onto a different cluster

navsetlur opened the pull request:
…rrent clusterID. If that's not the case, pause the scheduled changefeed until manual intervention

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/124631)
<!-- Reviewable:end -->

dt commented:
Looking good!

You'll want to amend the commit message (and PR title) to include a short title and then longer body that includes a release not, as laid out here: https://cockroachlabs.atlassian.net/wiki/spaces/CRDB/pages/73072807/Git+Commit+Messages#GitCommitMessages-Committitle 

I might suggest:
> changefeedccl: pause schedules when restored onto different cluster
>
> (some prose, that mentions that we do the same for scheduled backups)
> 
> Release note (enterprise change): The schedule for Scheduled Changefeeds now pauses after being restored onto a different cluster and after completion of physical cluster replication to avoid inadvertent concurrent execution of the same schedule on multiple clusters.
>
> Epic: none.


As far as testing, `TestPauseScheduledBackupOnNewClusterID` might be a useful source of inspiration.

navsetlur committed:
changefeedccl: pause schedules when restored onto a different cluster

In other schedules, such as for a restore, we pause the operation if the clusterID of the schedule differs from the clusterID of the current cluster.
This prevents unwanted operations on a replicated cluster until manual intervention is applied. This change adds the same check to scheduled changedfeeds.

Release note (enterprise change): The schedule for Scheduled Changefeeds now pauses after being restored onto a different cluster
and after completion of physical cluster replication to avoid inadvertent concurrent execution of the same schedule on multiple clusters.

Epic: none

navsetlur commented:
bors r=dt

navsetlur commented:
bors r=dt
